sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -864,34 +869,28 @@ ur_program_handle_t ProgramManager::getBuiltURProgram(
       for (RTDeviceBinaryImage *BinImg : DeviceImagesToLink) {
         if (UseDeviceLibs)
           DeviceLibReqMask |= getDeviceLibReqMask(*BinImg);
-        device_image_plain DevImagePlain =
-            getDeviceImageFromBinaryImage(BinImg, Context, Device);
-        const std::shared_ptr<detail::device_image_impl> &DeviceImageImpl =
-            detail::getSyclObjImpl(DevImagePlain);
 
-        SerializedObj ImgSpecConsts =
-            DeviceImageImpl->get_spec_const_blob_ref();
-
-        ur_program_handle_t NativePrg =
-            createURProgram(*BinImg, Context, {Device});
+        ur_program_handle_t NativePrg = createURProgram(*BinImg, Context, Devs);
 
         if (BinImg->supportsSpecConstants())
-          setSpecializationConstants(DeviceImageImpl, NativePrg, Adapter);
```

Comment:
Note: this spec constant part was intentionally removed since it did nothing except enable ITT annotations because the newly created device image contained no spec constant values.